File: fortran-tests/gemm-2.f90
subroutine gemm(m, n, k, alpha, A, B, beta, C)
  integer :: m, n, k
  real, dimension(m, n) :: C
  real, dimension(m, k) :: A
  real, dimension(k, n) :: B
  real :: alpha, beta
  real :: aa, bb, cc

  do mm = 1, m
    do nn = 1, n
      cc = 0.0
      do i = 1, k
        aa = A(mm, i)
        bb = B(nn, i)
        cc = cc + aa * bb;
      end do
      C(mm, nn) = C(mm, nn) * beta + alpha * cc
    end do
  end do
end subroutine

File: cplus-tests/gemm-2.cc
void basicSgemm(int m, int n, int k, float alpha, const float **A, const float **B, float beta, float **C )
{
  for (int mm = 0; mm < m; ++mm) {
    for (int nn = 0; nn < n; ++nn) {
      float c = 0.0f;
      for (int i = 0; i < k; ++i) {
        float a = A[mm][i];
        float b = B[nn][i];
        c += a * b;
      }
      C[mm][nn] = C[mm][nn] * beta + alpha * c;
    }
  }
}
